connect/runtime/src/main/java/org/apache/kafka/connect/runtime/standalone/StandaloneHerder.java

```diff
@@ -296,7 +301,12 @@ public synchronized void restartConnector(String connName, Callback<Void> cb) {
 
         startConnector(connName, (error, targetState) -> {
             if (targetState == TargetState.STARTED) {
-                requestTaskReconfiguration(connName);
+                try {
+                    requestTaskReconfiguration(connName);
+                } catch (Throwable t) {
+                    cb.onCompletion(t, null);
```

Comment:
Same here. Should we log it?

Reply:
Same as above, this also comes from a REST API call.